src/activity_handlers.cpp

```diff
@@ -1879,13 +1879,20 @@ void activity_handlers::reload_finish( player_activity *act, player *p )
     const int qty = act->index;
     const bool is_speedloader = ammo.has_flag( flag_SPEEDLOADER );
     const bool is_bolt = ammo.ammo_type() == ammo_bolt;
+    const bool reloadable_is_filthy = reloadable.is_filthy();
+    const bool ammo_is_filthy = ammo.is_filthy();
 
     if( !reloadable.reload( *p, std::move( act->targets[ 1 ] ), qty ) ) {
         add_msg( m_info, _( "Can't reload the %s." ), reloadable.tname() );
         return;
     }
 
     std::string msg = _( "You reload the %s." );
+
+    if( ammo_is_filthy && !reloadable_is_filthy ) {
+        reloadable.set_flag( "FILTHY" );
+    }
```

Comment:
Check for `!reloadable_is_filthy` is redundant. If it's already filthy, nothing changes.